service/notification/NotificationProcessor.php

```diff
@@ -129,11 +129,41 @@ public function processNotification($unprocessedNotification)
 
         switch ($unprocessedNotification['event_code']) {
             case AdyenNotification::AUTHORISATION:
-                $order->setCurrentState(\Configuration::get('PS_OS_PAYMENT'));
-                $this->orderService->addPaymentDataToOrderFromResponse($order, $unprocessedNotification);
+                if ('true' === $unprocessedNotification['success']) {
```

Comment:
looks a bit weird should we not move this to unprocessedNotification['success'] == true or does boolean checking not working ?

Reply:
should we not store it as well as a boolean value ?

Reply:
The database table structure indicates the value is a string and we would like to keep this check simple that's why we don't cast it to a boolean and just deal with it as a string. We could store it of course as a boolean in the future.

Reply:
ok let's keep it is like this for now.